File: src/main/java/com/vocalclerk/api/dtos/TaskDto.java
package com.vocalclerk.api.dtos;

import com.vocalclerk.domain.entities.TaskPriority;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

/**
 * DTO for transferring task data to and from the API.
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class TaskDto {
    private String id;
    private String title;
    private String description;
    private LocalDateTime dueDate;
    private LocalDateTime reminderTime;
    private boolean completed;
    private TaskPriority priority;
    private String categoryId;
    private String categoryName;
    private List<String> tags;
    private LocalDateTime createdAt;
    private LocalDateTime completedAt;
}

File: backend/DoNow.Application/DTOs/TaskDto.cs
using DoNow.Domain.Entities;

namespace DoNow.Application.DTOs;

public class TaskDto
{
    public long Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public DateTime? DueDate { get; set; }
    public DateTime? ReminderTime { get; set; }
    public bool Completed { get; set; }
    public TaskPriority Priority { get; set; }
    public long? CategoryId { get; set; }
    public string? CategoryName { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? CompletedAt { get; set; }
    public List<string> Tags { get; set; } = new();
}
